src/services/shell.service.js

```diff
@@ -2,18 +2,49 @@
 /**
  * Thin wrapper around electron `shell`, and other shell-like functions
  */
-import { shell } from 'electron';
+import { shell, remote } from 'electron';
 import launchEditor from 'react-dev-utils/launchEditor';
 import { exec } from 'child_process';
 
 import type { Project } from '../types';
 
+const { BrowserWindow } = remote;
+
+const openedWindows = {};
+
 export const openProjectInFolder = (project: Project) =>
   shell.openItem(project.path);
 
 export const openProjectInEditor = (project: Project) =>
   launchEditor(project.path, 1, 1);
 
+export const openWindow = (url: string) => {
+  const urlKey = url.replace(/(\/|:|\.)/g, '');
+
+  if (openedWindows[urlKey]) {
+    try {
+      return openedWindows[urlKey].show();
+    } catch (err) {
+      // swallow error - Window could be closed
+    }
+  }
+
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      devTools: false,
+    },
+  });
+  // Remove the menu
+  win.setMenu(null);
+  win.loadURL(url);
+  win.show();
+
+  // Save Window so we can open it later
+  openedWindows[urlKey] = win;
+};
```

Comment:
Hm so on my machine, it just opened a new chrome tab, the same way `shell.openExternal` works.

I think this is fine :) but yeah if you're keen to have it open a new app window, might need some debugging, at least on mac.

Reply:
Oh, OK, for me this is working and opens a new electron window. Just tested on Windows.
So if we're using it later, we have to check the behaviour on Mac.